### PR TITLE
Adding test for rgw pod restart in mcg_write_to_bucket

### PR DESCRIPTION
**Pull request by parikshithb**
This test runs rbd/cephfs and s3 bucket IOs and checks for rgw restarts if any.

### CODE REVIEW

File: tests/manage/mcg/test_write_to_bucket.py [hunk 1]
```diff
@@ -78,3 +104,54 @@ def test_data_reduction(self, mcg_obj, awscli_pod, bucket_factory):
...
+    @pytest.mark.polarion_id("OCS-2040")
+    def test_write_to_bucket_rbd_cephfs(self, setup_rbd_cephfs_pods, retrive_s3_objects,
+                                        mcg_obj, awscli_pod, bucket_factory
+                                        ):
+        """
+        Test RGW restarts after running s3, rbd and cephfs IOs in parallel
+
+        """
+        logger.info('RGW restart count before running IOs')
+        pods = get_rgw_pods()
```
Comment (hnallurv) [inline tests/manage/mcg/test_write_to_bucket.py]: how many rgw pods are supposed to be running in the cluster? If only one, no need for looping.

File: tests/manage/mcg/test_write_to_bucket.py [hunk 2]
```diff
@@ -78,3 +104,54 @@ def test_data_reduction(self, mcg_obj, awscli_pod, bucket_factory):
...
+        for rgw_pod in pods:
+            rgw_restart_count = rgw_pod.restart_count
```
Comment (hnallurv) [inline tests/manage/mcg/test_write_to_bucket.py]: Isn't rgw_restart_count getting overwritten with multiple rgw pods restart count here?

File: tests/manage/mcg/test_write_to_bucket.py [hunk 3]
```diff
@@ -78,3 +104,54 @@ def test_data_reduction(self, mcg_obj, awscli_pod, bucket_factory):
...
+
+        with ThreadPoolExecutor() as p:
+            p.submit(pod_io, setup_rbd_cephfs_pods)
+            p.submit(s3_io, retrive_s3_objects, mcg_obj, awscli_pod, bucket_factory)
+
+        logger.info("Checking whether RGW pod restarted")
+        for rgw_pod in pods:
+            rgw_pod.reload()
+            assert rgw_pod.restart_count == rgw_restart_count, 'RGW pod restarted after running parallel IOs'
```
Comment (hnallurv) [inline tests/manage/mcg/test_write_to_bucket.py]: Checking for rgw restart count should become one of the exit criteria for all noobaa based tests. Please make this as a common function and call it to verify the restart count in the applicable tests

File: ocs_ci/ocs/resources/mcg.py [hunk 4]
```diff
@@ -97,6 +98,19 @@ def __init__(self):
                 aws_access_key_id=self.aws_access_key_id,
                 aws_secret_access_key=self.aws_access_key
             )
+            logger.info('Checking whether RGW pod is not present on AWS platform')
+            pods = pod.get_rgw_pods()
+            assert len(pods) == 0, 'RGW pod is should not exist on AWS platform'
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py]: RGW pod should not exist on AWS platform

File: ocs_ci/ocs/resources/pod.py [hunk 5]
```diff
@@ -457,6 +461,25 @@ def get_csi_provisioner_pod(interface):
...
 
+def get_rgw_pods(rgw_label=constants.RGW_APP_LABEL, namespace=None):
+    """
+    Fetches info about rgw pods in the cluster
+
+    Args:
+        rgw_label (str): label associated with rgw pods
+            (default: defaults.RGW_APP_LABEL)
+        namespace (str): Namespace in which ceph cluster lives
+            (default: defaults.ROOK_CLUSTER_NAMESPACE)
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: the default is set to None, please fix

File: tests/manage/mcg/test_write_to_bucket.py [hunk 6]
```diff
@@ -19,26 +56,15 @@ class TestBucketIO(ManageTest):
     Test IO of a bucket
     """
     @pytest.mark.polarion_id("OCS-1300")
-    def test_write_file_to_bucket(self, mcg_obj, awscli_pod, bucket_factory, uploaded_objects):
+    def test_write_file_to_bucket(self, retrive_s3_objects, mcg_obj, awscli_pod, bucket_factory, uploaded_objects):
```
Comment (ebenahar) [inline tests/manage/mcg/test_write_to_bucket.py]: please break lines

File: tests/manage/mcg/test_write_to_bucket.py (hunk 1 above)
Comment (parikshithb) [reply tests/manage/mcg/test_write_to_bucket.py]: ack

File: tests/manage/mcg/test_write_to_bucket.py (hunk 2 above)
Comment (parikshithb) [reply tests/manage/mcg/test_write_to_bucket.py]: Removed the loop, getting the single rgw pod now.

File: ocs_ci/ocs/resources/pod.py (hunk 5 above)
Comment (parikshithb) [reply ocs_ci/ocs/resources/pod.py]: changed

File: tests/manage/mcg/test_write_to_bucket.py (hunk 6 above)
Comment (parikshithb) [reply tests/manage/mcg/test_write_to_bucket.py]: ack

File: ocs_ci/ocs/resources/mcg.py (hunk 4 above)
Comment (parikshithb) [reply ocs_ci/ocs/resources/mcg.py]: ack

File: tests/manage/mcg/test_write_to_bucket.py (hunk 3 above)
Comment (parikshithb) [reply tests/manage/mcg/test_write_to_bucket.py]: Added verify rgw pod restart count fixture to bucket creation and deletion tests. Mainly intended for performance tests which does 1000 bucket creation and deletion.

**Pull request merged**